### PR TITLE
perf: Apply various workarounds to mitigate PG-AOT dotnet/runtime issue #56309

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): https://github.com/dotnet/runtime/issues/56309

## PR Type

What kind of change does this PR introduce?
- Refactoring (no functional changes, no api changes)

## What is the new behavior?

These changes are required to reduce the interpreter performance hit.  The AOT compiler detects try/catch/finally containing methods and forces the interpreter, even if the try/catch is not in the execution path.

Inner methods are created to reduce the size of the interpreter-executed code, and return to AOT as soon as possible.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/DataBinding/BindingExpression.cs [hunk 1]
```diff
@@ -219,6 +210,27 @@ public void UpdateSource(object value)
 			}
 		}
 
+		/// <remarks>
+		/// This method contains or is called by a try/catch containing method and
+		/// can be significantly slower than other methods as a result on WebAssembly.
```
Comment (dr1rrb) [inline src/Uno.UI/DataBinding/BindingExpression.cs:215]: "and can" ???

**Review by dr1rrb — APPROVED** [review]

**Review by carldebilly — APPROVED** [review]
TL;JA

**Comment by ajpinedam** [comment]
> TL;JA

@carldebilly does this mean: Too long J'ai approved?

**Review by ajpinedam — APPROVED** [review]

**Comment by carldebilly** [comment]
> > TL;JA
> 
> @carldebilly does this mean: Too long J'ai approved?

Too Long; Just Approved ;-)

**Pull request merged**